### PR TITLE
added sitemap

### PR DESCRIPTION
**Pull request by Neobii**
This adds a sitemap and robots file in `postbuild` with [next-sitemap](https://github.com/iamvishnusankar/next-sitemap).

### CODE REVIEW

File: vulcan-next-sitemap.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
+  siteUrl: 'https://vulcan.next',
+  generateRobotsTxt: true,
+  //outDir: './out/', for static builds
+  exclude: ['/vns/*', '/login', '/profile', '/signup', '/admin', '/auth'],
```
Comment (eric-burel) [inline vulcan-next-sitemap.js:5]: Can we instead have a whitelist approach with `include`? Vulcan app are often private app so it make more sense to whitelist what you want to see in the sitemap than what you don't want to see

File: public/sitemap.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (eric-burel) [inline public/sitemap.xml:1]: If it is in the postbuild command, you might want to add this file to `.gitignore`

File: public/robots.txt [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# *
+User-agent: *
+Disallow: /vns/*
```
Comment (eric-burel) [inline public/robots.txt:3]: Is it also autogenerated? It could also be in .gitignore

**Comment by eric-burel** [comment]
Great stuff thank you! Would you mind adding a word about this feature in the documentation? The file is there: https://github.com/VulcanJS/vulcan-next/blob/devel/src/pages/docs/features.md

I try to keep track of all features we add progressively

File: vulcan-next-sitemap.js (hunk 1 above)
Comment (Neobii) [reply vulcan-next-sitemap.js:5]: It doesn't look like that package supports include at the moment.

Comment (Neobii) [reply vulcan-next-sitemap.js:5]: The robots.txt file I'm thinking of is something like?
```allow: ['/','/docs/*'],```

**Comment by eric-burel** [comment]
Sounds good to me thanks for the quick update and the work!

**Comment by Neobii** [comment]
No problem, should I write a test for this?

**Comment by eric-burel** [comment]
that could help indeed, you could add a test in "/tests/vns", for example inspired from "packages.test.ts". You could run the relevant `yarn` command as done in this example and check that the files are created as expected, I think that would be sufficient to tell that the sitemap script is working ok

**Pull request merged**